### PR TITLE
Fixes being unable to dump storage contents onto floors

### LINKED ISSUE #30007: Drag-and-drop to empty containers isn't happening.
Drag-and-drop to empty containers isn't happening. No messages in chat, nothing. Can't dump to floor.

Round ID 74877
Revision 093cb1a184f5a4d31ad71e8f522827f3493c5a69

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
fix: You can now empty storage objects onto floors again.
/:cl:

It seems like someone forgot to make floors' `get_dumping_location()` return itself, so I fixed this. I can tell this is unintentional because walls have an override despite `/atom` already returning null; I'm thus doubtful that this was removed intentionally and personally chalk it up to oversight.

Fixes #30007

### CODE REVIEW

**Comment by Hatfish** [comment]
Will this also fix dumping containers into other containers?

**Comment by Xhuis** [comment]
I did that during testing, and it seemed to work as intended.

**Comment by AnturK** [comment]
Oh whoops, must have forgotten to add this.

**Review by AnturK — APPROVED** [review]

File: code/game/turfs/simulated/floor.dm [hunk 1]
```diff
@@ -253,3 +253,6 @@
 			G.anchored = TRUE
 			return TRUE
 	return FALSE
+
+/turf/open/floor/get_dumping_location()
```
Comment (AnturK) [inline code/game/turfs/simulated/floor.dm]: Actually should probably be /open, don't see a reason why you shouldn't be able to dump in space.

**Comment by Cyberboss** [comment]
@AnturK I think this should be merged with drop_location. But I'm having a hard time determining what the use cases should be.. Say you had an apple in a bag

* Apple.drop_location() should return (the turf or the bag?)
* Bag.drop_location() should return (the turf or itself?)
* Turf.drop_location() should return itself
* Disposal.drop_location() should return itself (right?)

I'm trying to just nail down what is what so the system can function without confusion

**Comment by AnturK** [comment]
Probably can be merged, in all cases it's just first container that allows dropping and you can reach starting from your whatever you aimed at or your location for dropping. If there's no user that's just first container.

So assuming you aim for these and there's no reach issues: 
- Bag
- Itself
- Itself
- Itself
- Disposals

**Pull request merged**